### ISSUE TITLE
Uncaught InvalidStateError: Failed to execute 'drawImage' on 'CanvasRenderingContext2D': The imag...

### ISSUE THREAD

**Issue by krotera** [atom-minimap/minimap#681]
1. Have two files open in a vertically-split workspace
2. Drag the separator line to resize the workspace all the way to the right

**Atom**: 1.34.0 x64
**Electron**: 3.0.13
**OS**: "Manjaro Linux"
**Thrown From**: [minimap](https://github.com/atom-minimap/minimap) package 4.29.9

### Stack Trace

Uncaught InvalidStateError: Failed to execute 'drawImage' on 'CanvasRenderingContext2D': The image argument is a canvas element with a width or height of 0.

```
At /home/qio/.atom/packages/minimap/lib/canvas-layer.js:64

Error: Failed to execute 'drawImage' on 'CanvasRenderingContext2D': The image argument is a canvas element with a width or height of 0.
    at CanvasLayer.copyToOffscreen (/packages/minimap/lib/canvas-layer.js:64:27)
    at HTMLElement.redrawRangesOnLayer (/packages/minimap/lib/mixins/canvas-drawer.js:285:11)
    at HTMLElement.updateTokensLayer (/packages/minimap/lib/mixins/canvas-drawer.js:136:10)
    at HTMLElement.updateCanvas (/packages/minimap/lib/mixins/canvas-drawer.js:104:10)
    at HTMLElement.update (/packages/minimap/lib/minimap-element.js:887:10)
    at requestAnimationFrame (/packages/minimap/lib/minimap-element.js:780:12)
```

### Commands

```
     -2:33.9.0 core:move-left (input.hidden-input)
     -2:33.7.0 core:move-right (input.hidden-input)
     -2:33.6.0 intentions:highlight (input.hidden-input)
     -2:33.5.0 editor:select-to-beginning-of-word (input.hidden-input)
     -2:31.7.0 autocomplete-plus:cancel (input.hidden-input)
     -2:31.4.0 core:move-down (input.hidden-input)
  4x -2:31.2.0 core:move-left (input.hidden-input)
     -2:30.6.0 intentions:highlight (input.hidden-input)
     -2:30.5.0 editor:select-to-beginning-of-word (input.hidden-input)
     -2:28.4.0 intentions:highlight (input.hidden-input)
     -2:28.3.0 core:save (input.hidden-input)
  2x -2:23 core:move-left (input.hidden-input)
     -2:22.7.0 intentions:highlight (input.hidden-input)
     -2:22.6.0 editor:select-to-end-of-word (input.hidden-input)
     -2:21.4.0 intentions:highlight (input.hidden-input)
     -2:21.3.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
busy-signal 2.0.0 
duotone-dark-earth-syntax 2.1.0 
duotone-dark-forest-syntax 2.1.0 
duotone-dark-sea-syntax 2.1.0 
highlight-line 0.12.0 
highlight-selected 0.14.0 
intentions 1.1.5 
language-rust 0.4.12 
linter 2.2.0 
linter-gcc 0.7.1 
linter-rust 0.9.0 
linter-ui-default 1.7.1 
minimap 4.29.9 
minimap-find-and-replace 4.5.2 
minimap-highlight-selected 4.6.1 
```

**Comment by Amxx** [comment]
getting the same error

**Comment by farooque-abd** [comment]
Getting same error Over & over again.

**Comment by grace-grimwood** [comment]
Getting the same error on Windows 10, Atom 1.39.1 x64, Electron 3.1.10, Chrome 66.0.3359.181, Node v10.2.0.

**Comment by Amxx** [comment]
Anyone working on a fix?
Can't use the minimap package with that ...

**Comment by leovazquezz1** [comment]
Same issue here

**Comment by wanancat** [comment]
same error on macOS Mojave 10.14.6, Atom 1.40.1 x64.

**Comment by arsoljon** [comment]
same.

**Comment by rbianchini** [comment]
Hello, the same issue here.
$ atom --version
Atom    : 1.41.0
Electron: 4.2.7
Chrome  : 69.0.3497.128
Node    : 10.11.0
OS: Ubuntu 18.04.3 LTS

**Comment by DannyDannyDanny** [comment]
I get this when moving windows across multiple screens on W10 (using winkey + left).

$ atom 
Atom    : 1.41.0
Electron: 4.2.7
Chrome  : 69.0.3497.128
Node    : 10.11.0

**Comment by gonidelis** [comment]
same error

**Comment by Bhorlarh** [comment]
same error

**Comment by AlbertoFabbri93** [comment]
I got the same error on Windows 10 with Atom 1.42.0 x64

**Comment by binbjz** [comment]
same error on Mac os Catalina 10.15.2

**Comment by DavidWellsTheDeveloper** [comment]
Yup same issue Ubuntu 18.04. Great package other than this...

**Comment by ashthespy** [comment]
Ran into the same issue, while resizing a pane.

**Atom**: 1.44.0 x64
**Electron**: 4.2.7
**OS**: Ubuntu 19.04
**Thrown From**: [minimap](https://github.com/atom-minimap/minimap) package 4.29.9


### Stack Trace

Uncaught InvalidStateError: Failed to execute 'drawImage' on 'CanvasRenderingContext2D': The image argument is a canvas element with a width or height of 0.

```
At /home/default_user/.atom/packages/minimap/lib/canvas-layer.js:64

Error: Failed to execute 'drawImage' on 'CanvasRenderingContext2D': The image argument is a canvas element with a width or height of 0.
    at CanvasLayer.copyToOffscreen (/packages/minimap/lib/canvas-layer.js:64:27)
    at HTMLElement.redrawRangesOnLayer (/packages/minimap/lib/mixins/canvas-drawer.js:285:11)
    at HTMLElement.updateTokensLayer (/packages/minimap/lib/mixins/canvas-drawer.js:136:10)
    at HTMLElement.updateCanvas (/packages/minimap/lib/mixins/canvas-drawer.js:104:10)
    at HTMLElement.update (/packages/minimap/lib/minimap-element.js:887:10)
    at requestAnimationFrame (/packages/minimap/lib/minimap-element.js:780:12)
```

**Comment by Ks89** [comment]
Same issue on macOS catalina when double clicking on the window border. It happens randomly.

**Comment by aminya** [comment]
Duplicate of https://github.com/atom-minimap/minimap/issues/702. Will be fixed in #713
